provider/fs/src/export/serializers/mod.rs

```diff
@@ -6,42 +6,19 @@ pub mod bincode;
 pub mod json;
 pub mod postcard;
 
-use displaydoc::Display;
 use icu_provider::buf::BufferFormat;
+use icu_provider::prelude::*;
+use icu_provider::serde::SerializeMarker;
 use std::io;
 
-/// An Error type specifically for the [`Serializer`](serde::Serializer) that doesn't carry filenames
-#[derive(Display, Debug)]
-#[non_exhaustive]
-pub enum Error {
-    #[displaydoc("{0}")]
-    Io(io::Error),
-    #[displaydoc("{0}")]
-    Serializer(erased_serde::Error),
-}
-
-impl std::error::Error for Error {}
-
-impl From<io::Error> for Error {
-    fn from(e: io::Error) -> Self {
-        Error::Io(e)
-    }
-}
-
-impl From<erased_serde::Error> for Error {
-    fn from(e: erased_serde::Error) -> Self {
-        Error::Serializer(e)
-    }
-}
-
 /// A simple serializer trait that works on whole objects.
 pub trait AbstractSerializer {
     /// Serializes an object to a sink.
     fn serialize(
         &self,
-        obj: &dyn erased_serde::Serialize,
+        obj: DataPayload<SerializeMarker>,
```

Comment:
Nit (optional): since you don't need ownership, take `&DataPayload<SerializeMarker>` or `&SerializeBox` (the return value of `DataPayload::get()`)

Reply:
I'll leave it like this for now but I have a further change here in the queue.